include/zephyr/math/q_ops.h

```diff
@@ -0,0 +1,133 @@
+        	__ASSERT_NO_MSG(intermediate <= INT32_MAX && intermediate >= INT32_MIN);           \
+		qval.value = intermediate & 0xffffffff;                                            \
+	} while (false)
+
+#define q_add(qval, x)                                                                             \
+	do {                                                                                       \
+                int64_t lhs = (int64_t)qval.value;                                                 \
+                int64_t rhs = z_q_val_to_intermediate(x) >> z_q_right_shift_bits(qval);            \
+		int64_t intermediate = lhs + rhs;                                                  \
+        	__ASSERT_NO_MSG(intermediate <= INT32_MAX && intermediate >= INT32_MIN);           \
+		qval.value = intermediate & 0xffffffff;                                            \
+	} while (false)
+
+#define q_sub(qval, x)                                                                             \
+	do {                                                                                       \
+                int64_t lhs = (int64_t)qval.value;                                                 \
+                int64_t rhs = z_q_val_to_intermediate(x) >> z_q_right_shift_bits(qval);            \
+		int64_t intermediate = lhs - rhs;                                                  \
+        	__ASSERT_NO_MSG(intermediate <= INT32_MAX && intermediate >= INT32_MIN);           \
+		qval.value = intermediate & 0xffffffff;                                            \
+	} while (false)
+
+#define q_mul(qval, x)                                                                             \
+	do {                                                                                       \
+                int64_t lhs = (int64_t)qval.value;                                                 \
+		int64_t rhs = z_q_val_to_intermediate(x) >> z_q_right_shift_bits(qval);            \
+		int64_t intermediate =  (lhs * rhs) >> (32 - z_q_right_shift_bits(qval));          \
+        	__ASSERT_NO_MSG(intermediate <= INT32_MAX && intermediate >= INT32_MIN);           \
+		qval.value = intermediate & 0xffffffff;                                            \
+	} while (false)
+
+#define q_div(qval, x)                                                                             \
+	do {                                                                                       \
+		int64_t lhs = (int64_t)qval.value << (32 - z_q_right_shift_bits(qval));            \
+		int64_t rhs = z_q_val_to_intermediate(x) >> z_q_right_shift_bits(qval);            \
+		int64_t intermediate = lhs / rhs;                                                  \
+        	__ASSERT_NO_MSG(intermediate <= INT32_MAX && intermediate >= INT32_MIN);           \
+		qval.value = intermediate & 0xffffffff;                                            \
+	} while (false)
```

Comment:
Is there any reason to use false and depend of stdbool instead of just use while(0) ?

Reply:
Just the coding style